dowhy/causal_refuters/add_unobserved_common_cause.py

```diff
@@ -226,6 +244,48 @@ def refute_estimate(self, show_progress_bar=False):
 
             analyzer.check_sensitivity(plot=self.plot_estimate)
             return analyzer
+
+        if self.simulation_method == "non-parametric-partial-R2":
+            # If the estimator used is LinearDML, partially linear sensitivity analysis will be automatically chosen
+            if isinstance(self._estimate.estimator, dowhy.causal_estimators.econml.Econml):
```

Comment:
We don't have the architecture for it yet --- but an if condition such as this one --- that is checking to see what estimator was used --- makes me think that we should be prescribing the sensitivity analysis method at the same place in the code where we set the estimator method.  Perhaps the estimator object registers what sensitivity analysis method should be used, or perhaps both are registered together.  Spreading this logic across different parts of the code is going to cause trouble at some point.

This is just me thinking out loud.  I don't expect a change in this PR.

Reply:
great point, Emre. Yes, let's take it up when we discuss how to refactor refutations to work across the estimation pipeline.